src/Hedgehog.Experimental/Gen.fs

```diff
@@ -26,6 +26,32 @@
     let cString (lower : int) (upper : int) : (Gen<char> -> Gen<string>) =
         string (Range.constant lower upper)
 
+    /// Generates a string that is not equal to another string using
+    /// StringComparison.OrdinalIgnoreCase.
+    let iNotEqualTo (str : string) : (Gen<string> -> Gen<string>) =
```

Comment:
The idea for the generator itself sounds OK but I'm having a hard time to figure out the naming convention (now that it's been a while since I last looked at these generators).

How can we make the names more descriptive?

Reply:
I'm all in favor of generally using descriptive names, but as I have described, there needs to be a balance between descriptiveness and conciseness. Do we really need to make the names more descriptive in this specific case? Once users have looked at the docstring/intellisense, which explains that it's case insensitive, I think the names make good sense.

Reply:
What if we put them in nested modules?

Reply:
Doesn't solve anything; that would either force users to write the module name each time they use it (e.g. `CaseInsensitive.notEqualTo`), putting us back to start, or else `open` the modules and shadowing the default generators.

Reply:
Isn't CaseInsensitive.notEqualTo easier to read and reason about compared to iNotEqualTo? 

Not that I'm against it... I'm just trying to act as a random consumer of those (which isn't probably aware of this discussion).

In general, when I'm not sure about something, I invoke the [Zen of Python](https://www.python.org/dev/peps/pep-0020/) -- *Explicit is better than implicit.*

Let's see if @jystic has a better idea 😕

Reply:
Instead of `CaseInsensitive.notEqualTo` we could just as well call it `caseInsensitiveNotEqualTo` and skip the extra module. I just think the name is a bit too long to write all the time when I could be writing `iNotEqualTo`.

In any case, while I may have strong opinions on how long names I want to write regularly, don't let me dictate this decision. :)

Reply:
I've been going back and forth on this one; naming is hard! Should we play it safe and go with `caseInsensitiveNotEqualTo` and so on? Then we can go ahead an make a NuGet package 🎉

Reply:
Keep in mind we're in v0.* and this means we can rename those easily until we reach v1.

Reply:
As I've said, my money is on `i...` but I'll live with whatever you decide. Perhaps @jystic has some opinions on this?